plugins/chromatic_correction.py

```diff
@@ -0,0 +1,228 @@
+        :param data: value and metadata of the given data array
+        """
+        # save the image
+        dataio.tiff.export(filename, data)
+        # Read the saved images and display
+        analysis_tab = self.main_app.main_data.getTabByName('analysis')
+        self.main_app.main_data.tab.value = analysis_tab
+        analysis_tab.load_data(filename, extend=True)
+
+    def start(self):
+        localization_tab = self.main_app.main_data.getTabByName("cryosecom-localization")
+        tab_data = localization_tab.tab_data_model
+        other_mip_streams = []
+        reference_mip_stream = []
+        levels = generate_zlevels(tab_data.main.focus,
+                                  (-10e-06, 10e-06),
+                                  5e-06)
+        non_static = [s for s in tab_data.streams.value
+                      if isinstance(s, FluoStream)]
+        zlevels = {s: levels for s in non_static}
+        settings_observer = tab_data.main.settings_obs
+        acqui_task = acquireZStack(non_static, zlevels, settings_obs=settings_observer)
+        das, e = acqui_task.result()
+
+        if len(das) <= 1:
+            box = wx.MessageDialog(self.main_app.main_frame,
+                                   "Add minimum two streams. One of the streams must be green channel.",
+                                   "Failed to do chromatic correction", wx.OK | wx.ICON_STOP)
+            box.ShowModal()
+            box.Destroy()
+            return
+
+        # Calculate Maximum Intensity projection for all the present channels in the stream panel
+        for i, da in enumerate(das):
+            mip_image = model.DataArray(numpy.amax(da, axis=0), copy.copy(da.metadata))
+            dataio.tiff.export(f"{da.metadata[model.MD_DESCRIPTION]}_{i}", mip_image)
+            # All the channels will be corrected according to a reference colored channel
+            # green channel is selected as reference channel, Why? -> Ask Deniz
+            if da.metadata[model.MD_USER_TINT] == (0, 255, 0):
+                reference_mip_stream = mip_image
+            else:
+                other_mip_streams.append(mip_image)
+
+        # Save and display the reference channel in the analysis tab
+        self.show_static_stream(f"modified_{reference_mip_stream.metadata[model.MD_DESCRIPTION]}_ref",
+                                reference_mip_stream)
+
+        # Compute transformation between reference channel and other channels
+        trans_per_channel = {}  # stores transformation values per channel
+        _, centers_ref = process_image(reference_mip_stream)
+        centers_ref = numpy.array(centers_ref)
+        for i, im in enumerate(other_mip_streams):
+            # Find corresponding points
+            _, centers_new = process_image(im)
+            centers_new = numpy.array(centers_new)
+            max_distance = BEAD_DIAMETER
+            corresponding_pairs = find_corresponding_points(centers_ref, centers_new, max_distance)
```

Comment:
I'd suggest to adjust compute_transformation() to accept 2 parameters: spot_diameter, and max_distance.
The first one you use in MaximaFind, and the second one in `find_corresponding_points()`.
Then add a comment just before calling `compute_transformation()`  to explain why you use spot_diameter as a max_distance too. Or maybe for now you can use spot_diameter * 1.1 to already show that it can be different... but for now you want to keep it small to handle better the clusters.